src/SonarScanner.MSBuild.PreProcessor/Roslyn/RoslynAnalyzerProvider.cs

```diff
@@ -28,109 +28,101 @@
     }
 
-    private IEnumerable<string> WriteAdditionalFiles(string language, IEnumerable<SonarRule> activeRules)
+    private IEnumerable<string> WriteAdditionalFiles(IEnumerable<SonarRule> activeRules)
     {
-        Debug.Assert(activeRules != null, "Supplied active rules should not be null");
+        Debug.Assert(activeRules is not null, "Supplied active rules should not be null");
 
-        if (TryWriteSonarLintXmlFile(language, activeRules, out var filePath))
+        if (TryWriteSonarLintXmlFile(activeRules, out var filePath))
         {
             Debug.Assert(File.Exists(filePath), "Expecting the additional file to exist: {0}", filePath);
-            return new[] { filePath };
+            return [filePath];
         }
-
-        return Enumerable.Empty<string>();
+        return [];
     }
 
-    private bool TryWriteSonarLintXmlFile(string language, IEnumerable<SonarRule> activeRules, out string sonarLintXmlPath)
+    private bool TryWriteSonarLintXmlFile(IEnumerable<SonarRule> activeRules, out string sonarLintXmlPath)
     {
-        var langDir = Path.Combine(this.teamBuildSettings.SonarConfigDirectory, language);
+        var langDir = Path.Combine(teamBuildSettings.SonarConfigDirectory, language);
         Directory.CreateDirectory(langDir);
 
         sonarLintXmlPath = Path.Combine(langDir, "SonarLint.xml");
         if (File.Exists(sonarLintXmlPath))
         {
-            this.logger.LogDebug(Resources.RAP_AdditionalFileAlreadyExists, language, sonarLintXmlPath);
+            logger.LogDebug(Resources.RAP_AdditionalFileAlreadyExists, language, sonarLintXmlPath);
             return false;
         }
 
-        var content = RoslynSonarLint.GenerateXml(activeRules, this.sonarProperties, language);
-        this.logger.LogDebug(Resources.RAP_WritingAdditionalFile, sonarLintXmlPath);
+        var content = RoslynSonarLint.GenerateXml(activeRules, sonarProperties, language);
+        logger.LogDebug(Resources.RAP_WritingAdditionalFile, sonarLintXmlPath);
         File.WriteAllText(sonarLintXmlPath, content);
         return true;
     }
 
-    private IEnumerable<AnalyzerPlugin> FetchAnalyzerPlugins(string language, IEnumerable<SonarRule> activeRules)
+    private IEnumerable<AnalyzerPlugin> FetchAnalyzerPlugins(IEnumerable<SonarRule> activeRules)
     {
         var partialRepoKeys = ActiveRulesPartialRepoKeys(activeRules);
-        IList<Plugin> plugins = new List<Plugin>();
+        IList<Plugin> plugins = [];
 
         foreach (var partialRepoKey in partialRepoKeys)
         {
-            if (!this.sonarProperties.TryGetValue($"{partialRepoKey}.pluginKey", out var pluginKey) ||
-                !this.sonarProperties.TryGetValue($"{partialRepoKey}.pluginVersion", out var pluginVersion) ||
-                !this.sonarProperties.TryGetValue($"{partialRepoKey}.staticResourceName", out var staticResourceName))
+            if (!sonarProperties.TryGetValue($"{partialRepoKey}.pluginKey", out var pluginKey) ||
```

Comment:
While this is nicer, the `||` needs to go at the beginning of the line.

Reply:
And that, together with the stupid `continue`, illustrates the problem with negative logic of this code. Make it positive.

Reply:
It's sooo much better now